packages/experimental/Text/src/Text.tsx

```diff
@@ -1,18 +1,93 @@
-import { textName, ITextType } from './Text.types';
-import { compose } from '@uifabricshared/foundation-compose';
+/** @jsx withSlots */
+import {
+  compose,
+  FontTokens,
+  FontVariantTokens,
+  UseSlots,
+  buildProps,
+  fontStyles,
+  mergeProps,
+  withSlots,
+  ITheme
+} from '@fluentui-react-native/experimental-framework';
 import { Text as RNText } from 'react-native';
-import { filterTextProps } from '@fluentui-react-native/adapters';
-import { foregroundColorTokens, textTokens } from '@fluentui-react-native/tokens';
-import { settings } from './Text.settings';
+import { filterTextProps, ITextProps } from '@fluentui-react-native/adapters';
 
-export const Text = compose<ITextType>({
+const textName = 'Text';
+
+/**
+ * Text tokens, these are the internally configurable values for Text elements. In particular these
+ * drive decisions on how to build the styles
+ */
+export interface TextTokens extends FontTokens {
+  /** foreground text color */
+  color?: string;
+
+  /** alternate mode for disabled look and feel */
+  disabled?: TextTokens;
```

Comment:
I'm unsure what this is for. Is it used to define a style for disabled? I'm confused why TextTokens is used in the definition of TextTokens.

Reply:
Yes, this is to define a style for disabled. The reason it is recursive (as a pattern) is that if there are multiple states they each can contain the other ones. (This way you can do hovered + pressed, or primary + disabled, if those are states you have.)

Reply:
This is equivalent to what before was:

```ts
settings: {
  _overrides: {
    disabled : {
      tokens: {
        // same content here
      }
    }
  }
```